odoo/addons/base/models/ir_actions_report.py

```diff
@@ -524,12 +525,22 @@ def _run_wkhtmltopdf(
 
         files_command_args = []
         temporary_files = []
+        temp_session = None
 
         # Passing the cookie to wkhtmltopdf in order to resolve internal links.
-        if request and request.db:
+        if request and request.db and request.session.sid:
```

Comment:
In what cases `request.session.sid` is falsy?

Reply:
It is possible to encounter this during testing when using `MockRequest`. The session will be a (dot)dict and will not contain `sid`.
In practice, this only affects one test, so I will modify the test in question to remove this part of the condition.